demos/coreHTTP/http_demo_s3_download_multithreaded.c

```diff
@@ -0,0 +1,1087 @@
+ * in this example.
+ * */
+static void prvStartHTTPTask( void * pvArgs );
+
+/**
+ * @brief Clean up resources created by demo.
+ *
+ * @param[in] xHandle The HTTP task handle.
+ * @param[in] xRequestQueue The request queue.
+ * @param[in] xResponseQueue The response queue.
+ */
+static void prvTearDown( TaskHandle_t xHandle,
+                         QueueHandle_t xRequestQueue,
+                         QueueHandle_t xResponseQueue );
+
+/*-----------------------------------------------------------*/
+
+static BaseType_t prvConnectToServer( NetworkContext_t * pxNetworkContext )
+{
+    ServerInfo_t xServerInfo = { 0 };
+    SocketsConfig_t xSocketsConfig = { 0 };
+    BaseType_t xStatus = pdPASS;
+    TransportSocketStatus_t xNetworkStatus = TRANSPORT_SOCKET_STATUS_SUCCESS;
+
+    configASSERT( pxNetworkContext != NULL );
+
+    /* Initializer server information. */
+    xServerInfo.pHostName = cServerHost;
+    xServerInfo.hostNameLength = xServerHostLength;
+    xServerInfo.port = democonfigHTTPS_PORT;
+
+    /* Configure credentials for TLS server-authenticated session. */
+    xSocketsConfig.enableTls = true;
+    xSocketsConfig.pAlpnProtos = NULL;
+    xSocketsConfig.maxFragmentLength = 0;
+    xSocketsConfig.disableSni = false;
+    xSocketsConfig.pRootCa = democonfigROOT_CA_PEM;
+    xSocketsConfig.rootCaSize = sizeof( democonfigROOT_CA_PEM );
+    xSocketsConfig.sendTimeoutMs = democonfigTRANSPORT_SEND_RECV_TIMEOUT_MS;
+    xSocketsConfig.recvTimeoutMs = democonfigTRANSPORT_SEND_RECV_TIMEOUT_MS;
+
+    /* Establish a TLS session with the HTTP server. This example connects
+     * to the server host located in democonfigPRESIGNED_GET_URL and
+     * democonfigHTTPS_PORT in http_s3_download_multithreaded_demo_config.h. */
+    LogInfo( ( "Establishing a TLS session with %s:%d.",
+               cServerHost,
+               democonfigHTTPS_PORT ) );
+
+    /* Attempt to create a server-authenticated TLS connection. */
+    xNetworkStatus = SecureSocketsTransport_Connect( pxNetworkContext,
+                                                     &xServerInfo,
+                                                     &xSocketsConfig );
+
+    if( xNetworkStatus != TRANSPORT_SOCKET_STATUS_SUCCESS )
+    {
+        LogWarn( ( "Unsuccessful connection attempt, received error code:%d",
+                   ( int ) xNetworkStatus ) );
+        xStatus = pdFAIL;
+    }
+
+    return xStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static BaseType_t prvDownloadS3ObjectFile( const char * pcHost,
+                                           const size_t xHostLen,
+                                           const char * pcRequest,
+                                           const size_t xRequestUriLen,
+                                           QueueHandle_t xRequestQueue,
+                                           QueueHandle_t xResponseQueue )
+{
+    BaseType_t xStatus = pdPASS;
+    size_t xResponseCount = 0;
```

Comment:
xResponseCount --> xRemainingResponseCount 

Would it be a better name?

Reply:
Sure, sounds good.